test/azul_test_case.py

```diff
@@ -1,30 +1,57 @@
+import logging
 import os
+from typing import List
 from unittest import TestCase
 from unittest.mock import (
     patch,
 )
 import warnings
+import re
 
 import boto3.session
 from botocore.credentials import Credentials
 import botocore.session
 
+log = logging.getLogger(__name__)
+
 
 class AzulTestCase(TestCase):
     _warning_state = None
+    _warning_log = List[warnings.WarningMessage]
+    _permitted_warnings = {DeprecationWarning: [re.compile('(.*?)')],
+                           ResourceWarning: [re.compile('unclosed.*<ssl.SSLSocket.*>')]}
 
     @classmethod
     def setUpClass(cls) -> None:
         super().setUpClass()
-        cls._warning_state = warnings.catch_warnings()
-        cls._warning_state.__enter__()
-        warnings.filterwarnings('ignore', 'unclosed.*<ssl.SSLSocket.*>', ResourceWarning)
+        cls._warning_state = warnings.catch_warnings(record=True)
+        cls._warning_log = cls._warning_state.__enter__()
 
     @classmethod
     def tearDownClass(cls) -> None:
+        if cls._inspect_for_bad_warnings():
+            pass
+            # at this point the test should already have a status failed or pass, how  to modify the test that occured?
```

Comment:
@hannes-ucsc attempting to fail at this point does not seem to work as expected due to the result of the test already being determined.
Is there a way to modify the result of the testcase that has already finished?